enterprise/frontend/src/metabase-enterprise/auth/actions.ts

```diff
@@ -1,12 +1,20 @@
-import { createThunkAction } from "metabase/lib/redux";
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { getSetting } from "metabase/selectors/settings";
+import { State } from "metabase-types/store";
 import { trackLoginSSO } from "./analytics";
 import { getSSOUrl } from "./utils";
 
+interface ThunkConfig {
+  state: State;
+}
+
 export const LOGIN_SSO = "metabase-enterprise/auth/LOGIN_SSO";
-export const loginSSO = createThunkAction(
+export const loginSSO = createAsyncThunk<void, string | undefined, ThunkConfig>(
```

Comment:
It would be more convenient to do it like [this](https://github.com/metabase/metabase/pull/30867/files#diff-c0273e494e96db6a97012f496e84c566f0604c71fd2db1de30f277a7edc6f696) — feel free to implement it in your PR because it will get merged earlier